rskj-core/src/main/java/org/ethereum/rpc/exception/RskErrorResolver.java

```diff
@@ -23,10 +25,21 @@ public JsonError resolveError(Throwable t, Method method, List<JsonNode> argumen
             error =  new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
         } else if (t instanceof InvalidFormatException) {
             error = new JsonError(-32603, "Internal server error, probably due to invalid parameter type", null);
+        } else if (t instanceof UnrecognizedPropertyException) {
+            String message = getExceptionMessage((UnrecognizedPropertyException) t);
+            logger.error("JsonRPC error {} for method {} with arguments {}", message, method, arguments);
+            error = new JsonError(-32603, message, null);
         } else {
             logger.error("JsonRPC error when for method {} with arguments {}", method, arguments, t);
             error = new JsonError(-32603, "Internal server error", null);
         }
         return error;
     }
+
+    private String getExceptionMessage(UnrecognizedPropertyException ex) {
```

Comment:
Maybe we can just use Throwable (which already provides the `getMessage` method) and avoid the cast?

Reply:
In the end I've opted for doing this cast since I need the exception to extract some data.